oketest.sh

```diff
@@ -0,0 +1,183 @@
+#!/bin/bash
+# Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+#
+# This script provisions a OKE Kubernetes cluster using terraform (https://www.terraform.io/) and runs the new
+# integration test suite against that cluster. Blog https://blogs.oracle.com/weblogicserver/easily-create-an-oci-container-engine-for-kubernetes-cluster-with-terraform-installer-to-run-weblogic-server
+# provides detailed explanation for OCI properties file creation.
+#
+#
+# As of May 6, 2020, the tests are clean on Kubernetes 1.16 with the following JDK workarounds:
+# 1. Maven must be run with OpenJDK 11.0.7, available here: https://github.com/AdoptOpenJDK/openjdk11-upstream-binaries/releases/download/jdk-11.0.7%2B10/OpenJDK11U-jdk_x64_linux_11.0.7_10.tar.gz
+#    This is because of a critical bug fix. Unfortunately, the Oracle JDK 11.0.7 release was based on an earlier build and doesn't have the fix.
+# 2. The WebLogic Image Tool will not accept an OpenJDK JDK. Set WIT_JAVA_HOME to an Oracle JDK Java Home.
+#    For example, "export WIT_JAVA_HOME=/usr/java/jdk-11.0.7" before running this script.
+#
+set -o errexit
+set -o pipefail
+
+script="${BASH_SOURCE[0]}"
+scriptDir="$( cd "$( dirname "${script}" )" && pwd )"
+
+function usage {
+  echo "usage: ${script} [-n <terraform config files directory>] [-o <directory>] [-t <tests>] [-c <name>] [-p true|false] [-x <number_of_threads>] [-d <wdt_download_url>] [-i <wit_download_url>] [-m <maven_profile_name>] [-h]"
+  echo "  -n Terraform config files directory "
+  echo "  -o Output directory (optional) "
+  echo "      (default: \${WORKSPACE}/logdir/\${BUILD_TAG}, if \${WORKSPACE} defined, else /scratch/\${USER}/kindtest) "
+  echo "  -b  Availability Domain Name "
+  echo "      (for example: VPGL:PHX-AD-1 , check limits quota with OCI admin)"
+  echo "  -t Test filter (optional) "
+  echo "      (default: **/It*) "
+  echo "  -s Oracle Cloud Infra properties file  "
+  echo "  -p Run It classes in parallel"
+  echo "      (default: false) "
+  echo "  -x Number of threads to run the classes in parallel"
+  echo "      (default: 2) "
+  echo "  -d WDT download URL"
+  echo "      (default: https://github.com/oracle/weblogic-deploy-tooling/releases/latest) "
+  echo "  -i WIT download URL"
+  echo "      (default: https://github.com/oracle/weblogic-image-tool/releases/latest) "
+  echo "  -m Run integration-tests or oke-cert "
+  echo "      (default: integration-tests, supported values: oke-cert) "
+  echo "  -h Help"
+  exit $1
+}
+
+function prop {
+    grep "${1}" ${oci_property_file}| grep -v "#" | cut -d'=' -f2
+}
+
+if [[ -z "${WORKSPACE}" ]]; then
+  outdir="/scratch/${USER}/oketest"
+  export WORKSPACE=${PWD}
+else
+  outdir="${WORKSPACE}/logdir/${BUILD_TAG}"
+fi
+test_filter="**/It*"
+parallel_run="false"
+threads="2"
```

Comment:
Can we make the number of thread as variable when we run the tests in parallel.

Reply:
it is optional variable, you can pass it as opt , this is just default value